### ISSUE TITLE
Rule: No control characters in regular expressions

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#338]
This rule should warn when a control character is found inside of a regular expression. For example:

``` js
var a1 = /\\\x1f/;
var a2 = /[\\\x1f]/;
```

We need to figure out which control characters end up as warnings. By definition, control characters are codes 1-31 and 127, but we should double-check which of these JSHint is checking.

The rule ID should be `no-regex-control-chars` and should be on by default.
